csrc/multidevice/communication.cpp

```diff
@@ -40,56 +42,148 @@ inline void assertBuffersHaveSameSize(
   }
 }
 
-inline void post_common(Communication& self, Communicator& comm) {
-  NVF_ERROR(
-      std::find(
-          self.params().team.begin(),
-          self.params().team.end(),
-          comm.deviceId()) != self.params().team.end(),
-      "current device index ",
-      comm.deviceId(),
-      " must be present in the communication's team");
-}
-
 inline void doLocalCopy(const at::Tensor& dst, const at::Tensor& src) {
   dst.copy_(src, /* non-blocking */ true);
 }
 
-} // namespace
+inline bool hasRoot(CommunicationType type) {
+  return type == CommunicationType::Gather ||
+      type == CommunicationType::Scatter ||
+      type == CommunicationType::Broadcast ||
+      type == CommunicationType::SendRecv;
+}
+
+inline bool isReduction(CommunicationType type) {
+  return type == CommunicationType::Reduce ||
+      type == CommunicationType::Allreduce ||
+      type == CommunicationType::ReduceScatter;
+}
+
+inline std::string typeToString(CommunicationType type) {
```

Comment:
I think it's sufficient to directly define an `operator<<` for `CommunicationType` without a typeToString helper, e.g., https://github.com/NVIDIA/Fuser/blob/4e8356915fc6ffa823cf97fed4249cc00f317fdf/csrc/type.cpp#L1231. `NVF_ERROR` is able to take a `operator<<`able type without using an explicit typeToString function.